providers/file_input.rb

```diff
@@ -28,6 +30,8 @@
               'state_file' => new_resource.name,
               'severity' => new_resource.severity,
               'facility' => new_resource.facility
-    notifies :restart, resources(:service => 'rsyslog')
+    notifies :restart, resources('service[rsyslog]')
   end
+
+  new_resource.updated_by_last_action(t.updated_by_last_action?)
```

Comment:
In an LWRP, when you use `use_inline_resources`, a local run context is built, and it is not scoped with the the resource collection from the rest of the Chef run, so when you try to notify the resource from the recipe, without having defined it in the LWRP, it will fail.